packages/@uppy/companion/src/server/controllers/s3.js

```diff
@@ -152,8 +169,14 @@ module.exports = function s3 (config) {
      */
     const client = req.companion.s3Client
     const { uploadId } = req.params
+    const bucket = req.query.bucket ?? config.bucket
```

Comment:
That seems quite dangerous to blindly trust the user input, I would expect some validation. https://github.com/transloadit/uppy/pull/4488 was suggestion a safer approach.